### PR TITLE
Do not register an offense when using try with an enumerable accessor

### PR DESCRIPTION
**Pull request by rrosenblum**
I realized that I did not account for enumerable accessors in the `SafeNavigation` cops. I decided to not have them register an offense for now. As far as I know, safe navigation cannot be used directly on them. `foo&.[0]` is invalid. `foo&.public_send(:[], 0)` would work, but I am not sure that we want to suggest that as a replacement or not.

I will also need to account for this in `Style/SafeNavigation`.

### CODE REVIEW

**Comment by bbatsov** [comment]
Apart from the failing build, the changes look good to me.

**Comment by rrosenblum** [comment]
I also remembered that I may need to handle assignment methods slightly differently than what I am currently doing. I will update this PR when I get a chance. I plan on looking through some of the test cases for other cops to see if I can think of any other edge cases that we have run into in the past.

**Comment by rrosenblum** [comment]
I made a few tweaks to help prevent some bugs. This will now not register an offense for operators and variables passed as the first argument to `try!`. The regex being used to ignore operators is a bit aggressive and will also ignore any method that you wouldn't normally call using `.` (`[], +, -, +=`).

```ruby
[5] pry(main)> foo = 5
=> 5
[6] pry(main)> foo.== 5
=> true
[7] pry(main)> foo. == 5
=> true
[8] pry(main)> foo&.== 5
=> true
[9] pry(main)> foo&. == 5
=> true
```

File: lib/rubocop/cop/rails/safe_navigation.rb [hunk 1]
```diff
@@ -42,30 +44,38 @@ class SafeNavigation < Cop
...
           return if target_ruby_version < 2.3
 
-          try_call(node) do |method|
-            return if method == :try && !cop_config['ConvertTry']
-            add_offense(node, :expression, format(MSG, method))
+          try_call(node) do |try_method, method_to_try|
+            return if try_method == :try && !cop_config['ConvertTry']
+            return unless method_to_try.sym_type?
+            method, = *method_to_try
+            return unless method =~ /\w+[=!?]?/
```
Comment (rrosenblum) [inline lib/rubocop/cop/rails/safe_navigation.rb]: Does anyone know if there is regex similar to this that exists in another cop? It is matching against standard methods that are comprised of words, underscores, and end with `?`, `!`, or `=`.
